src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -236,20 +199,55 @@ public Task WriteEndBatchAsync()
         }
 #endif
 
-        /// <summary>Starts a new changeset; can only be called after WriteStartBatch and if no other active operation or changeset exists.</summary>
+        /// <summary>
+        /// Starts a new changeset without specifying group id.
+        /// This can only be called after WriteStartBatch and if no other active operation or changeset exists.
+        /// </summary>
         public void WriteStartChangeset()
+        {
+            this.WriteStartChangeset(null);

```

Comment:
I would pass Guid.NetGuid().ToString to this.WriteStartChangeset.  Then, in the WriteStartChangesetImplementation for JsonLight you don't need to account for a null groupId, and in multipartmixed you can generate the changeset boundary using that groupid.

Reply:
OK. This requires other refactoring of the ODataMultipartMixedBatchWriterUtils.CreateChangeSetBoundary as well.

---
In reply to: [155383582](https://github.com/OData/odata.net/pull/992#discussion_r155383582) [](ancestors = 155383582)